src/saving/serializers/ThriveJsonConverter.cs

```diff
@@ -686,6 +686,11 @@ private void RunPostPropertyDeserializeActions(object instance)
             throw new JsonException($"no suitable constructor found for current type: {objectType}");
         }
 
+        if ( item == null)
+        {
+            throw new ArgumentNullException($" Non existing key from JObject: {item} ");
```

Comment:
Umm, the issue is about *catching* in the various converters like GodotColourConverter, GodotBasisConverter etc. and not about throwing here. And btw `item` is already used on line 670 so this check against null here is not very useful.